xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -508,17 +498,20 @@ void updatePriorityState(int priority) {
           }
         }
 
-        if (priority == currentPriority || overallState == READY) {
+        if (priority == currentPriority) {
+          updatePicker(overallState, childPickers);
+          if (overallState == READY) {
+            cancelFailOverTimer();
+            goToPriority(priority);
```

Comment:
This is not necessary.

Reply:
This is necessary, because `currentPriority` after the change is a temporary value. That's why I didn't like `currentPriority`  to be temporary value cross multiple methods.

Reply:
Now inlined `goToPriority()`.